modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -2098,7 +2098,14 @@ protected CompletableFuture<ClusterNode> evaluateReadOnlyRecipientNode(int partI
                         throw withCause(TransactionException::new, REPLICA_UNAVAILABLE_ERR, e);
                     } else {
                         if (res == null) {
-                            throw withCause(TransactionException::new, REPLICA_UNAVAILABLE_ERR, e);
+                            throw new TransactionException(
```

Comment:
I suggest making it a separate method/constructor/static method so as not to clutter the code.